python_modules/dagster/dagster/_core/definitions/schedule_definition.py

```diff
@@ -736,12 +753,65 @@ def evaluate_tick(self, context: "ScheduleEvaluationContext") -> ScheduleExecuti
             skip_message = None
 
         # clone all the run requests with the required schedule tags
-        run_requests_with_schedule_tags = [
-            request.with_replaced_attrs(
-                tags=merge_dicts(request.tags, DagsterRun.tags_for_schedule(self))
+        run_requests_with_schedule_tags = []
+
+        for run_request in run_requests:
+            if run_request.partition_key and not run_request.has_resolved_partition_run_request():
+                if context.repository_def is None:
+                    raise DagsterInvariantViolationError(
+                        "Must provide repository def to build_schedule_context when yielding"
+                        " partitioned run requests"
+                    )
+
+                scheduled_target = (
+                    self.load_target()
+                    if isinstance(self._target, DirectTarget)
+                    else context.repository_def.get_job(self._target.pipeline_name)
+                )
+
+                if isinstance(scheduled_target, GraphDefinition):
+                    raise Exception(
+                        f"Error in schedule {self._name}: Schedule returned a partitioned"
+                        " RunRequest, but the target is a graph with no partitions definition."
+                    )
+
+                if isinstance(scheduled_target, PipelineDefinition):
+                    if scheduled_target.is_job:
```

Comment:
shouldn't this be

```
if not scheduled_target.is_job:
```

but then i don't understand why tests wouldn't be failing

Reply:
ah good catch, yes the `not` is supposed to be in there. the tests didn't fail because this code path is only hit when `partition_key` is specified on the run request, and there wasn't an existing test for that. I've added a test now